app/analysis/bikelanes.sql

```diff
@@ -0,0 +1,91 @@
+-- this function quantizes `LineStrings` into `Points` where each point represents a homogenous segment each of length  `res` +/- 25% (Note: segments shorter than `res` keep their length)
+CREATE OR REPLACE FUNCTION QuantizeLineString(ls Geometry(LineString), res numeric)
+    RETURNS TABLE(
+        len numeric,
+        geom Geometry(point)
+    )
+    AS $$
+DECLARE
+    len numeric := st_length(ls);
+    n int := greatest(round(len / res), 1)::int;
+BEGIN
+    RETURN query
+    SELECT
+        len / n,
+        ST_LineInterpolatePoint(ls,(generate_series(1, n) - 0.5) / n::numeric) AS geom;
+END;
+$$
+LANGUAGE plpgsql;
+
+-- for `jsonb` with numeric values return a normalized object which also includes the total value
+CREATE OR REPLACE FUNCTION NormalizeDistribution(dist jsonb)
+    RETURNS jsonb
+    AS $$
+DECLARE
+    total numeric := sum(value)
+FROM (
+    SELECT
+        (jsonb_each(dist)).value::numeric) AS sq;
+BEGIN
+    RETURN jsonb_object_agg(key, round((100 * value::numeric) / total, 2)) || jsonb_build_object('total', total)
+FROM (
+    SELECT
+        (jsonb_each(dist)).*) AS sq;
+END;
+$$
+LANGUAGE plpgsql;
+
+-- STEP 1: quantize bikelane data into points
+CREATE temp TABLE _bikelanesQuantized AS
+SELECT
+    osm_id,
+    tags,
+(QuantizeLineString(geom, 100)).* AS geom
+FROM
+    bikelanes;
+
+-- create a geospatial index on `_bikelanesQuantized` to speed up the spatial join;
+CREATE INDEX "_bikelanes_stats_geom_idx" ON _bikelanesQuantized USING gist(geom);
+
+-- STEP 2: calculate the distribution of our `category` for each area of the `boundary` data set
+-- make `osm_id` the primary of the `boundaries` table to allow selecting non-group-by columns
+ALTER TABLE boundaries
+    DROP CONSTRAINT IF EXISTS osm_id_key;
+
+ALTER TABLE boundaries
+    ADD CONSTRAINT osm_id_key PRIMARY KEY (osm_id);
+
+-- spatialy join `_bikelanesQuantized` with `boundaries` then group by category and boundaries.osm_id aggreagate the results in a single json object per area
+DROP TABLE IF EXISTS "boundaryStats";
+
+CREATE TABLE "boundaryStats" AS
+SELECT
+    osm_id,
+    geom,
+    region,
+    NormalizeDistribution(jsonb_object_agg(category, len)) AS stats
+FROM (
+    SELECT
+        boundaries.osm_id AS osm_id,
+        boundaries.geom AS geom,
+        boundaries.tags ->> 'name' AS region,
+        _bikelanesQuantized.tags ->> 'category' AS category,
+        sum(_bikelanesQuantized.len) AS len
+    FROM
+        boundaries
+        JOIN _bikelanesQuantized ON ST_Intersects(boundaries.geom, _bikelanesQuantized.geom)
+    WHERE (boundaries.tags ->> 'admin_level')::int BETWEEN 10 AND 10
```

Comment:
Laut https://wiki.openstreetmap.org/wiki/DE:Grenze#Innerstaatliche_Grenzen erfasst `admin_level=4` alle Bundesländer, was ich stichprobenhaft bestätigen konnte.

```suggestion
    WHERE (boundaries.tags ->> 'admin_level')::int = 4
```

Die Grafik auf der Seite zeigt auch, dass wir noch andere Grenzen anschauen könnten. Aber das würde ich später machen, da noch nicht klar ist, wie wir das anzeigen würden. Wenn ich das richtig sehe, währe es dann ein `admin_level IN (4, 6, 7, 8)` da 6-8 die Landkreise/Ämter, etc darstellt, die wir im Atlas meist darstellen.

```suggestion
    WHERE (boundaries.tags ->> 'admin_level')::int IN (4, 6, 7, 8)
```

Wenn du also Performance und edge cases testen willst, können wir auch das erstmal nutzen.